Qa development

bdrum opened the pull request:
Hello, Marian.

I rewrote the code.
1. Remove: GetSelector(), CheckProperty(), CountObjects();
2. For IsSelected() modificator changed to protected. It use only in GetProperty(), seems the best way put it inside of GetProperty(). I will think about optimisation of this process;
3. GetProperty() check all style and take last matched;
4. For <Class>ApplyStyle I deleted input parameter number of object and change something, now it's more fast and readable;
4. AliDrawStyleTest passed;

Hope it's better now, but I see some places which could be better. I will try to improvement their.

bdrum committed:
ATO-418: Bug fixes and improvements:
            1. Added check property function. Now if property doesn't
            exist we don't touch this.
            2. Fixed seg.v. when pad doesn't exist;
            3. Changed the rest IntegerAt to FloatAt for sizes and width.
            4. Added declaration in for loop for AliGenPerformance::Generate().

bdrum committed:
Merge remote-tracking branch 'refs/remotes/miranov25/QA-development' into QA-development

bdrum committed:
ATO-418: AliDrawStyle test improvements:
         1. Now it's more automaticaly

bdrum committed:
ATO-418: Rewriting code according to css standarts:
         1. Remove: GetSelector(), CheckProperty(), CountObjects();
         2. For IsSelected() modificator changed to protected. It use only in
            GetProperty(), seems the best way put it inside of GetProperty(). I will think
            about optimisation of this process;
         3. GetProperty() check all style and take last matched;
         4. AliDrawStyleTest passed;

miranov25 commented on EVGEN/AliGenPerformance.cxx:
```diff
@@ -14,28 +14,28 @@
 
 // Generator for particles according generic functions
-// For high pt performance studies realistic distribution of the particles - local density in jets to be approximated 
+// For high pt performance studies realistic distribution of the particles - local density in jets to be approximated
```
TO use official version

miranov25 commented on STAT/test/statTest.sh:
```diff
@@ -70,10 +70,21 @@ testAliDrawStyleTest() {
     gSystem->AddIncludePath("-I$ALICE_ROOT/include");
     .x ./AliDrawStyleTest.C+
 EOF
+
```
Should be hardwired or more save logic

miranov25 commented on STAT/AliDrawStyle.h:
```diff
@@ -62,15 +62,13 @@ class AliDrawStyle : public TObject{
-  static void TGraphApplyStyle(const char* styleName, TGraph *tempGraph, TString elementName, TString className, TString objName, Int_t objNum);
-  static void TH1ApplyStyle(const char* styleName, TH1 *tempHis, TString elementName, TString className, TString objName, Int_t objNum);
-  static void TF1ApplyStyle(const char* styleName, TF1 *tempFunc, TString elementName, TString className, TString objName, Int_t objNum);
+  static void TGraphApplyStyle(const char* styleName, TGraph *tempGraph, TString elementName, TString className, TString objName);
```
maybe remove temp prefix

miranov25 commented on STAT/AliDrawStyle.h:
```diff
@@ -85,7 +83,6 @@ class AliDrawStyle : public TObject{
   static TString  GetPropertyValue(TString input, TString propertyName);
   static Int_t    GetNamedIntegerAt(TString input, TString propertyName, Int_t index);
   static Float_t  GetNamedFloatAt(TString input, TString propertyName, Int_t index);
-  static Bool_t  IsSelected(TString selectors, TString elementName, TString className, TString objectName);
```
Return back to public

miranov25 commented on STAT/AliDrawStyle.cxx:
```diff
@@ -653,63 +661,52 @@ AliDrawStyle::GetProperty("alirootTestStyle.css","marker_size", "TGraph", "Statu
+  TPRegexp objPat(".*[[?]|.*[.]class|.*");
+  objNum = TString(TString(objName(numPat0))(numPat1)).Atoi();
+  objName = TString(objName(objPat)).ReplaceAll("[", "").ReplaceAll(".class", "");
+  TString pMarkerColor = AliDrawStyle::GetProperty(styleName, "marker_color", elementName, className, objName);
```
Example modification for color:

Bool_t status;
TString pMarkerColor = AliDrawStyle::GetProperty(styleName, "marker_color", elementName, className, objName);
value=AliDrawStyle::GetNamedFloatAt(pMarkerSize, "", objNum,status);
if (status)graph->SetMarkerStyle(value);

bdrum committed:
ATO-418: Some improvements

bdrum committed:
ATO-418: Changes:
         1. statTest.sh revert back;
         2. AliDrawStyleTest.C: added test for ApplyCssStyle, but not
         invariant;
         3. ApplyStyleTest not working with the empty classID;
